src/data/dfinsupp/basic.lean

```diff
@@ -1929,3 +1929,17 @@ add_monoid_hom.congr_fun (comp_lift_add_hom h.to_add_monoid_hom g) f
 end add_equiv
 
 end
+
+section finite_infinite
+
+noncomputable instance dfinsupp.fintype {ι : Sort*} {π : ι → Sort*} [fintype ι]
+  [∀ i, fintype (π i)] [Π i, has_zero (π i)] :
+  fintype (Π₀ i, π i) :=
+by letI := classical.dec_eq ι; exact fintype.of_injective _ fun_like.coe_injective
```

Comment:
This one should be definitely  be constructive, as pretty much everything about `dfinsupp` is.